### PR TITLE
Fix clients ignoring deletions that result in empty months, among smaller bugs

### PR DESCRIPTION
**Pull request by arneturuc**
### Clients ignoring deletions
Clients only overwrite data files when new data arrives.
They ignore files for which no data arrives, even though that is used to signal that the data for that file has been deleted completely.
Now, all data files are deleted before getting rebuild, ensuring that only the data present on the server is present on the client (aside from a possible open interval).

### Notification for active tracking
The notification for active time tracking is rephrased and moved, as to not give the impression that there was a gap and to hint at possible conflicts between open and closed intervals.

### Ignoring invalid data sources
When collecting data source files, `re.search` is replaced by `re.fullmatch` to ensure that only valid files (`YYYY-MM.data`) are included.
This should prevent similarly-named files (e.g. `2024-02.data.bak`) from being collected and/or deleted.

### CODE REVIEW

**Review by joachimschmidt557 — APPROVED** [review]
Looks good to me, apart from one minor warning message change.

File: timewsync/__init__.py [hunk 1]
```diff
@@ -230,19 +234,10 @@ def sync(configuration: Configuration) -> None:
...
-    if active_interval:
-        if started_tracking:
-            print("Restarted time tracking.", file=sys.stderr)
-        else:
-            log.warning(
-                "Warning: Cannot restart time tracking! This error occured because there exists "
-                "an interval in the future which would overlap with the open interval."
-            )
+    if active_interval and not started_tracking:
+        log.warning("Warning: Cannot restart time tracking because there exists a time interval in the future!")
```
[the remark this answers is not in the document]
Comment (arneturuc) [reply timewsync/__init__.py:240]:
```suggestion
        log.warning(
            "Warning: Cannot restart time tracking because there exists a time interval in the future "
            "which would overlap with the open interval!"
        )
```

**Review by joachimschmidt557 — APPROVED** [review]

File: timewsync/__init__.py [hunk 2]
```diff
@@ -230,19 +234,13 @@ def sync(configuration: Configuration) -> None:
...
-        if started_tracking:
-            print("Restarted time tracking.", file=sys.stderr)
-        else:
-            log.warning(
-                "Warning: Cannot restart time tracking! This error occured because there exists "
-                "an interval in the future which would overlap with the open interval."
-            )
+    if active_interval and not started_tracking:
+        log.warning(
+            "Warning: Cannot restart time tracking because there exists a time interval in the future "
```
Comment (arneturuc) [inline timewsync/__init__.py:241]: The text `Warning` should be redundant since we log it accordingly anyways.
```suggestion
            "Cannot restart time tracking because there exists a time interval in the future "
```

File: timewsync/__init__.py [hunk 3]
```diff
@@ -184,6 +184,10 @@ def sync(configuration: Configuration) -> None:
         log.error("Unexpected error occurred during JWT generation. No changes were made.")
         return
 
+    # Active time tracking
+    if active_interval:
+        print("Time tracking is active. Stopped and restarted time tracking to prevent conflicts.", file=sys.stderr)
```
Comment (vstollen) [inline timewsync/__init__.py:189]: Before the "restart" was only part of the message if we actually were able to restart the tracking. Is this change deliberate?

If not we should add a check for `started_tracking`.

**Review by joachimschmidt557 — APPROVED** [review]

**Review by vstollen — APPROVED** [review]
Looks good to me. We might want to add tests for the fixed bugs.

File: timewsync/__init__.py (hunk 3 above)
Comment (arneturuc) [reply timewsync/__init__.py:189]: That's a good point, maybe the previous approach was better in that regard!
Will update the message a little

**Comment by arneturuc** [comment]
> Looks good to me. We might want to add tests for the fixed bugs.

For me the problem lies in that both bugs depend on the data files, which I can't simulate. Will think about this

**Comment by vstollen** [comment]
Fair, we could probably mock the file system calls for the regex fix.

The other test feels more appropriate for an integration test (which we currently don't have).
